processing/src/main/java/org/apache/carbondata/processing/loading/converter/impl/RowConverterImpl.java

```diff
@@ -162,28 +167,96 @@ public DictionaryClient call() throws Exception {
     return null;
   }
 
+  private int getDataFieldIndexByName(String column) {
+    for (int i = 0; i < fields.length; i++) {
+      if (fields[i].getColumn().getColName().equalsIgnoreCase(column)) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  private String generateNonSchemaColumnValue(DataField field, CarbonRow row) {
```

Comment:
why this method is here. Better to handle at Field Converter layer. Index column is will take input from other columns and  generate the data . Changing code in RowConverter is not clean

Reply:
Agreed. Have defined a new FieldConverter for index column(IndexFieldConverterImpl). And this code is removed from this file.